### PR TITLE
Some updates

### PR DESCRIPTION
**Pull request by chilampoon**
Overlay ReactomeGSA results on the diagrams, and some minor updates

### CODE REVIEW

File: vignettes/ReactomeContentService4R.Rmd [hunk 1]
```diff
@@ -231,19 +231,62 @@ getOrthology("R-SSC-69541", species = "human")
...
+
+```{r ReactomeGSA, eval=FALSE}
+library(ReactomeGSA)
+library(ReactomeGSA.data)
+data("griss_melanoma_proteomics")
+
+# Create an analysis request and set parameters
+my_request <- ReactomeAnalysisRequest(method = "Camera")
+my_request <- set_parameters(request = my_request, max_missing_values = 0.5)
+my_request <- add_dataset(request = my_request,
```
Comment (cookersjs) [inline vignettes/ReactomeContentService4R.Rmd:246]: I have trouble running this command, and I think its related to the installlation of ReactomeGSA that I have. I followed the bioconductor instructions but I get this error: 

```
Error in (function (classes, fdef, mtable)  : 
  unable to find an inherited method for function ‘add_dataset’ for signature ‘"ReactomeAnalysisRequest", "character"’
```

Comment (chilampoon) [reply vignettes/ReactomeContentService4R.Rmd:246]: Have you installed `ReactomeGSA.data`?

Comment (cookersjs) [reply vignettes/ReactomeContentService4R.Rmd:246]: Yes, using `BiocManager::install("ReactomeGSA.data")`. Looking over the install logs, it looks like there was an issue installing/compiling 'mgcv'

Comment (chilampoon) [reply vignettes/ReactomeContentService4R.Rmd:246]: Oh `ReactomeGSA.data` may fail to install

File: vignettes/ReactomeContentService4R.Rmd [hunk 2]
```diff
@@ -231,19 +231,62 @@ getOrthology("R-SSC-69541", species = "human")
...
+                          expression_values = griss_melanoma_proteomics, 
+                          name = "Proteomics", 
+                          type = "proteomics_int",
+                          comparison_factor = "condition", 
+                          comparison_group_1 = "MOCK", 
+                          comparison_group_2 = "MCM",
+                          additional_factors = c("cell.type", "patient.id"))
+
+# Run analysis
+result <- perform_reactome_analysis(request = my_request, compress = F)
```
Comment (cookersjs) [inline vignettes/ReactomeContentService4R.Rmd:256]: Error with this part after I installed the less recent version of mgcv:

```
> result <- perform_reactome_analysis(request = my_request, compress = F)
Submitting request to Reactome API...
Error in curl::curl_fetch_memory(url, handle = handle) : 
  HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1)
```

File: vignettes/ReactomeContentService4R.Rmd [hunk 3]
```diff
@@ -231,19 +231,62 @@ getOrthology("R-SSC-69541", species = "human")
...
 
-```{r gif}
-gif <- exportImage(id = "R-HSA-69278", output = "diagram", format = "gif",
-                   sel = "R-HSA-69242", token = "MjAyMDA2MTcyMDM5NDBfMzU2")
-print(gif)
+The [ReactomeGSA](https://github.com/reactome/ReactomeGSA) package is an R client to the web-based Reactome Analysis Service, so we could perform analysis in R then access the token.
+
+```{r ReactomeGSA, eval=FALSE}
+library(ReactomeGSA)
+library(ReactomeGSA.data)
```
Comment (cookersjs) [inline vignettes/ReactomeContentService4R.Rmd:240]: 'mgcv' gave compilation issues. The solution is to download it manually and not install sources from package that needs compilation:

`> install.packages("mgcv")`
`Do you want to install from sources the package which needs compilation? (Yes/no/cancel) no`

This allowed me to get past the 'my_request' examples, but still fails at the 'result' part

File: vignettes/ReactomeContentService4R.Rmd [hunk 4]
```diff
@@ -231,19 +231,62 @@ getOrthology("R-SSC-69541", species = "human")
...
+data("griss_melanoma_proteomics")
```
[the remark this answers is not in the document]
Comment (chilampoon) [reply vignettes/ReactomeContentService4R.Rmd:241]: Actually just `data("griss_melanoma_proteomics")` then there will be an object `griss_melanoma_proteomics` in the environment

[the remark this answers is not in the document]
Comment (cookersjs) [reply vignettes/ReactomeContentService4R.Rmd:241]: Oh, actually when I installed/restarted and tried it now with the 'mgcv' library installed (non-compiled) and using your original code the result query worked!

File: vignettes/ReactomeContentService4R.Rmd (hunk 2 above)
Comment (cookersjs) [reply vignettes/ReactomeContentService4R.Rmd:256]: Works now!

**Pull request merged**